src/frequenz/sdk/timeseries/formula_engine/_formula_generators/_battery_power_formula.py

```diff
@@ -65,39 +69,106 @@ def generate(
 
         component_graph = connection_manager.get().component_graph
 
-        battery_inverters = frozenset(
-            frozenset(
+        # all_connected_batteries has to be set because
+        # multiple inverters can be connected to the same battery
+        inv_bat_mapping: dict[Component, set[Component]] = {}
+        all_connected_batteries: set[Component] = set()
```

Comment:
`all_connected_batteries` appears to have been a per-battery variable, to identify all the sibling batteries to any given battery.  Maybe it was not clear from the name.

It shouldn't be moved outside of the `for bat_id in component_ids:` loop, because it does not serve its purpose in that level.

Reply:
Done